coordinate-systems.md

```diff
@@ -0,0 +1,246 @@
+    *   frame: the frame for which the results are valid. This is needed as the current API shape requires XRFrame objects to resolve transforms of hits and the device
+    *   coordinate system: represents the coordinate system to which the hits are relative. This is needed as hit-results are most likely expressed in native _eye-level_ space and this parameter gives the app the information it needs to convert them into other coordinate systems.
+
+This API shape is extremely complicated. If we are always applying the device pose in the target FOR to the camera, why do we need to know where hit-results are in anything other than absolute space? We can always calculate vectors from the device to the hit in absolute space by using the device pose we get from `xrFrame.getDevicePose('stage');` or equivalent. If we want to render a reticle, we should similarly be placing our virtual reticle object in the app's absolute space as the camera is already accounting for the target reference frame. Furthermore, the underlying AR tracking system expresses hits in its native space, equivalent to eye-level, and thus those hit-results already match the implicit connection described above between the app's absolute space and the native tracking system's absolute space.
+
+If we assume the XRFrameOfReference and XRFrame API changes proposed above, we can re-simplify the hit-test API (and use a promise again!):
+
+
+```
+Promise<FrozenArray<XRHitResult>> requestHitTest(XRRay ray, XRCoordinateSystem coordinateSystem);
+const hitTransform = XRHitResult.hitMatrix;
+```
+
+
+We still need an API for getting a dynamic XRCoordinateSystem for the device, and we definitely need to have the transform of the target FOR baked in so that we get a ray relative to the right device pose. For example, we need the ray to be transformed 10 meters forward if we are in _stage_ and have transported as in the example used previously.
+
+In order to do that we could change the API for device pose to look like this:
+
+
+```
+XRCoordinateSystem XRSession.getDevicePose('stage');
+```
+
+
+Hold on - let's review the public API on XRCoordinateSystem:
+
+Float32Array XRCoordinateSystem::getTransformTo(XRCoordinateSystem* other);
+
+This isn't actually what we need for the hit-test API at all. We really just want something like XRDevicePose, which just has a poseModelMatrix property and it would be implied that the matrix would represent a transform in the current FOR mode. Under the hood, the hit-test logic would update the device pose for the new frame, compose the ray with the updated poseModelMatrix for the device, and then perform the hit-test calculation and return the results in absolute space. The API shape would then be:
```

Comment:
In WebXR 1.0, if the goal is to specifically restrict platform raycasts to `XRDevice` and `XRInputSource`, we could also just put the `requestHitTest` method on those types directly.  This avoids the dependency on bending another type like `XRDevicePose` to be common across to input sources, while leaving our design space more open for future open-ended raycast APIs for apps that have requested world-understanding permissions from the user.